packages/expo-module-autolinking/src/index.ts

```diff
@@ -0,0 +1,103 @@
+import commander from 'commander';
+
+import {
+  findModulesAsync,
+  resolveModulesAsync,
+  verifySearchResults,
+  generatePackageListAsync,
+  mergeLinkingOptionsAsync,
+} from './autolinking';
+import { GenerateOptions, ResolveOptions, SearchOptions, SearchResults } from './types';
+
+/**
+ * Registers a command that only searches for available expo modules.
+ */
+function registerSearchCommand<OptionsType extends SearchOptions>(
+  commandName: string,
+  fn: (search: SearchResults, options: OptionsType) => any
+) {
+  return commander
+    .command(`${commandName} [paths...]`)
+    .option<string[] | null>(
+      '-i, --ignore-paths <ignorePaths...>',
+      'Paths to ignore when looking up for modules.',
+      (value, previous) => (previous ?? []).concat(value)
+    )
+    .option<string[] | null>(
+      '-e, --exclude <exclude...>',
+      'Package names to exclude when looking up for modules.',
+      (value, previous) => (previous ?? []).concat(value)
+    )
+    .action(async (searchPaths, providedOptions) => {
+      const options = await mergeLinkingOptionsAsync<OptionsType>({
+        ...providedOptions,
+        searchPaths,
+      });
+      const searchResults = await findModulesAsync(options);
+      return await fn(searchResults, options);
+    });
+}
+
+/**
+ * Registers a command that searches for modules and then resolves them for specific platform.
+ */
+function registerResolveCommand<OptionsType extends ResolveOptions>(
+  commandName: string,
+  fn: (search: SearchResults, options: OptionsType) => any
+) {
+  return registerSearchCommand<OptionsType>(commandName, fn).option(
+    '-p, --platform [platform]',
+    'The platform that the resulted modules must support. Available options: "ios", "android"',
+    'ios'
+  );
+}
+
+module.exports = async function(args: string[]) {
+  // Searches for available expo modules.
+  registerSearchCommand('search', async results => {
+    console.log(require('util').inspect(results, false, null, true));
+  });
+
+  // Checks whether there are no resolving issues in the current setup.
+  registerSearchCommand('verify', results => {
+    const numberOfDuplicates = verifySearchResults(results);
+    if (!numberOfDuplicates) {
+      console.log('✅ Everything is fine!');
+    }
```

Comment:
It ok to inform about correct state, but what about the other case? 🤔 
Do the user/developer know which modules are duplicated across the project? 🤔 
I believe it might/should be provided alongside with some hints how to fix the problem 💪 🧰 💪

Reply:
In the other case, some logs will be printed by `verifySearchResults` so the user will know which ones are duplicated. Also, this is not being run by CocoaPods/Gradle scripts — its purpose is more for debugging and for people who pass custom search paths, which hopefully won't be needed by anyone 😄
Anyway, it's worth mentioning in the docs!